File: Java/src/Protocol/Room.java
/**
 * Mein Chatserver
 * ▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔▔
 * Licensed Materials - Property of mein-chatserver.de.
 * © Copyright 2024. All Rights Reserved.
 *
 * @version 1.0.0
 * @author  Adrian Preuß
 */

package Protocol;

import java.util.List;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Room implements IPacket {
	@JsonProperty("id")
    private int id;

    @JsonProperty("name")
    private String name;
    
    @JsonProperty("category")
    private Category category;

    @JsonProperty("style")
    private Style style;
    
    @JsonProperty("users")
    private List<User> users;
    
    public Room() {}

    public Room(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public int getID() {
        return id;
    }

    public void setID(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
    
    public List<User> getUsers() {
    	return this.users;
    }
    
    public Style getStyle() {
    	return this. style;
    }

    @Override
    public String toString() {
        return "[Room ID=" + id + ", Name=\"" + name + "\" ]";
    }

	@Override
	public String getOperation() {
		return "ROOMS";
	}
}

File: Angular/src/Models/Room.ts
import {Category} from './Category';
import {RoomConfiguration} from './RoomConfiguration';
import {RoomStyle} from './RoomStyle';
import {User} from './User';

export class Room {
  category: Category | null = null;
  id: number | null = null;
  name: string | null = null;
  topic: string | null = null;
  configuration: RoomConfiguration | null = null;
  users: User[] = [];
  style: RoomStyle | null = null;

  constructor(data: any) {
    if(data.category) {
      this.category = data.category;
    }

    if(data.id) {
      this.id = data.id;
    }

    if(data.name) {
      this.name = data.name;
    }

    if(data.topic) {
      this.topic = data.topic;
    }

    if(data.configuration) {
      this.configuration = data.configuration;
    }

    if(data.users) {
      this.users = data.users as User[];
    }

    if(data.style) {
      this.style = data.style as RoomStyle;
    }
  }

  getCategory() {
    return this.category;
  }

  getId() {
    return this.id;
  }

  getName() {
    return this.name;
  }

  getTopic() {
    return this.topic;
  }

  getConfiguration() {
    return this.configuration;
  }

  getUserCount() {
    return this.users.length;
  }
}
